frontend_build/src/apiSpecExportTools.js

```diff
@@ -0,0 +1,81 @@
+var fs = require("fs");
+var path = require("path");
+
+/**
+ * The following code is designed to read our apiSpec Javascript, but without having to resolve the
+ * requires contained therein, which include references to files that are not amenable to a vanilla
+ * node js require - however, they are properly handled by our webpack build process.
+ */
+
+// Find the API specification file relative to this file.
+var specFilePath = path.resolve(path.join(__dirname, '../../kolibri/core/assets/src/core-app/apiSpec.js'))
+
+// Read the spec file and do a regex replace to change all instances of 'require('...')'
+// to just be the string of the require path.
+var apiSpecFile = fs.readFileSync(specFilePath, 'utf-8').replace(/require\(('\S+')\)/g, '$1');
+
+// Invoke the module constructor to compile a module from this altered representation.
+var Module = module.constructor;
+
+var m = new Module(specFilePath, module.parent);
+
+m._compile(apiSpecFile, specFilePath);
+
+// Tada! The apiSpec object is now exported without doing any of the internal requires.
+var apiSpec = m.exports.apiSpec;
+
```

Comment:
Perhaps remove this empty line, and the ones between the lines in the previous "section", as it might help to visually group the sections together?